### PR TITLE
feat: dynamicaly update range based on provided data [DET-3923]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

- Dynamically figure out the max range of chart based on the data points. Add 10% padding to each axis for the max range.
- Handle the case of when new data points come into A) preserve the zoom if previously zoomed in or B) resize the chart to show all the data points including the new data points.

Screen Shot 2020-08-20 at 2 25 22 PM

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by armandmcqueen** [comment]
The code looks pretty good. A couple of minor issues:
- Because there is no line representing the x-axis, it's easy to confuse the gridlines for the x axis and then it looks a little strange when a data point is below that line
Screen Shot 2020-08-25 at 8 08 38 AM

- When I zoom in, there is no way to go back to the state where the ranges update as new data points come in. I can double click to zoom back out, but the scale no longer automatically adjusts.

File: webui/react/src/components/MetricChart.tsx [hunk 1]
```diff
@@ -76,38 +118,40 @@ const MetricChart: React.FC<Props> = (props: Props) => {
...
 
-  const handleScaleSelect = useCallback((newValue: SelectValue) => {
-    setScale(newValue as Scale);
-  }, []);
+  /*
+   * Dynamcially swapping out chart handlers is needed otherwise
+   * referenced data such as `maxRange` within the handlers will be stale.
+   */
+  useEffect(() => {
+    const chart = document.getElementById(id) as PlotlyHTMLElement;
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/components/MetricChart.tsx:145]: yep, good call!

**Comment by hkang1** [comment]
> The code looks pretty good. A couple of minor issues:
> 
> * Because there is no line representing the x-axis, it's easy to confuse the gridlines for the x axis and then it looks a little strange when a data point is below that line
> 
This one is tricky to address. Open to suggestions. Plotly renders the y and x axis (the zeroes) as the dark black lines, so hoping users will quickly learn to identify the zeroes. If the x and y axis were permanently in the view, it can skew the chart ranges depending on how far away from zero the x or the y values are clustered.

> * When I zoom in, there is no way to go back to the state where the ranges update as new data points come in. I can double click to zoom back out, but the scale no longer automatically adjusts.
>
Oops the `maxRange` should really be a state so when data points changed it will auto update (instead of just being a memoized value). This behavior should be fixed now (the chart, when not zoomed in, will auto adjust to show all the new data points).

**Comment by hkang1** [comment]
> Code looked to me too. Just this one that @armandmcqueen already pointed out. It was easy to repro with mnist_pytorch example
> 
> > When I zoom in, there is no way to go back to the state where the ranges update as new data points come in. I can double click to zoom back out, but the scale no longer automatically adjusts.
> 
> the chart goes out of the view when new points come in
> Screen Capture_select-area_20200825133657

Thanks for testing this, this should be corrected now.

**Comment by hkang1** [comment]
@armandmcqueen @hamidzr 
Thanks for trying out this PR out so thoroughly!

**Pull request merged**